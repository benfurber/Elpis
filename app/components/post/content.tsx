import React from "react";
import { StyleSheet, Text, View } from "react-native";
import FlexImage from "react-native-flex-image";

import { Post } from "interfaces";
import { colours, elements, layout } from "styles";
import { formatDate, validURL } from "utils";

interface Props {
  content: Post["content"];
  createdAt?: Post["createdAt"];
  imagePath: Post["imagePath"];
}

const Content = (props: Props) => {
  const renderText = () => {
    if (props.content) {
      return <Text style={styles.text}>{props.content}</Text>;
    }
  };

  const renderDate = () => {
    if (props.createdAt) {
      return (
        <Text style={[elements.textDate, styles.text]}>
          {formatDate(props.createdAt)}
        </Text>
      );
    }
  };

  const renderImage = () => {
    const { imagePath } = props;

    if (imagePath && validURL(imagePath)) {
      return <FlexImage source={{ uri: imagePath }} style={styles.image} />;
    }
  };

  return (
    <View style={styles.body}>
      {renderImage()}
      {renderText()}
      {renderDate()}
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: colours.whiteTransparent,
    borderTopRightRadius: layout.borderRadius,
    padding: layout.spacing,
    width: "100%",
  },
  text: {
    marginVertical: 10,
  },
  image: {
    borderRadius: layout.borderRadius,
    overflow: "hidden",
  },
});

export { Content };
