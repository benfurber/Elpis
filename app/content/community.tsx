import React from "react";
import { View, StyleSheet, Text } from "react-native";

import { Icon, Title } from "components";
import { labels } from "labels";
import { colours, layout } from "styles";

type SectionProps = {
  sectionText: {
    body: string;
    heading: string;
    subHeading: string;
  };
};

const content = [
  {
    body:
      "Bem-vinda à Elpis - um ambiente digital seguro onde você pode discutir tópicos importantes para você e obter suporte de outras usuárias da comunidade. Com sua participação ativa, nossas discussões podem ser um ótimo espaço para compartilhar experiências e opiniões pessoais e, finalmente, obter uma noção mais ampla do que outras usuárias como você vivenciam. Nosso desejo é que você encontre suporte e informações úteis e se divirta ao longo do caminho. Como em qualquer plataforma, precisamos seguir algumas regras. Familiarize-se com as diretrizes abaixo, que ajudarão a manter as coisas seguras e agradáveis ​​para todas nós. ",
    heading: "Introdução",
    subHeading: null,
  },
  {
    body:
      "As usuárias da plataforma falam sobre muitos tópicos que podem (ou não) ser sensíveis. Entendemos que existem abordagens diferentes para os mais diversos assuntos e as pessoas também podem ter opiniões diferentes. Mas lembre-se que sempre existe uma pessoa real do outro lado da tela lendo o que você escreve, que pode precisar do seu apoio e compreensão. Tenha empatia e seja sensível. Pense no que você gostaria (ou não) de ouvir. Se não tiver nada de bom para dizer, não diga nada. Vamos nos respeitar e ter conversas que exercitem o que temos de melhor enquanto comunidade.",
    heading: "Dicas básicas que todo membro deve seguir",
    subHeading: "Todas nós somos humanas.",
  },
  {
    body:
      "Tudo o que as usuárias e participantes da comunidade publicam no espaço está sendo lido e moderado. Nós trabalhamos constantemente para manter o equilíbrio entre a liberdade de expressão e a sensação de segurança de cada mulher aqui. Os debates são permitidos enquanto os diálogos permanecerem no tópico, positivos e respeitosos, sem focar em questões pessoais ou ataques entre as pessoas. Incentivamos  a comunicação não-violenta, o compartilhamento de experiências pessoais e a troca construtiva de opiniões.",
    subHeading: "Seja civilizada e respeitosa.",
  },
  {
    body:
      "Quando quiser pedir conselhos ou compartilhar sua opinião, examine os grupos e os tópicos e escolha o local apropriado para a publicação. Dessa forma, seu comentário será descoberto por outras usuárias muito mais rapidamente e você receberá comentários relevantes e úteis! Isso também ajuda a comunidade a se manter organizada. Se você não fizer isso, seu comentário poderá ser considerado fora do tópico e excluído pelos moderadores. E lembre-se de que os comentários aqui não são conselhos profissionais. Se você tiver alguma dúvida sobre sua saúde (física ou psicológica), é melhor consultar um profissional que possa cuidar do seu caso específico.",
    subHeading: "Mantenha-o relevante.",
  },
  {
    body:
      "Queremos enfatizar que o objetivo principal dos debates é oferecer a oportunidade única de conversar abertamente sobre tópicos relacionados às vidas das mulheres, encontrar apoio e aprender algo novo. A usuária que estiver associada a uma das condutas proibidas - descritas abaixo - estará passível de punição dentro do aplicativo (podendo ser banida da comunidade):",
    heading: "Discursos de ódio serão banidos",
  },
  {
    body:
      "Não toleramos racismo, sexismo, LGBTfobia, capacitismo, etarismo ou outras formas de discurso de ódio e preconceito. O ódio não tem lugar em nossa comunidade - estamos aqui para encontrar e oferecer apoio, compartilhar nossas experiências pessoais e nos ajudarmos, não para sermos julgadas ou criticadas. Quaisquer comentários que promovam ou incentivem a violência contra indivíduos ou grupos com base em raça ou etnia, crenças religiosas, deficiência, sexo, idade, nacionalidade, orientação sexual, identidade de gênero, visões políticas, estado de saúde ou aparência estarão sujeitos a exclusão.",
    subHeading: "Discurso de ódio não é aceito.",
  },
  {
    body:
      "Aqui, mulheres brasileiras de diversos lugares e culturas compartilham suas experiências e falam sobre tópicos que são importantes para elas. Isso significa que os comentários que revelam as informações pessoais de alguém, incitam assédio, ameaças, ataques pessoais ou apresentam comportamento abusivo ou predatório são levados muito a sério e estarão sujeitos a exclusão.",
    subHeading: "Bullying e ataques pessoais são proibidos.",
  },
  {
    body:
      "Encorajamos a publicação do tópico mais apropriado possível. Se você deseja pedir conselhos, procure as perguntas e respostas de outras usuárias sobre tópicos relevantes dentro do post em questão. Sua dúvida, questão ou problema já pode ter sido discutido - e você pode entrar no tópico em questão e postar uma resposta. Cada tópico tem um objetivo específico, que define as expectativas dos membros da comunidade sobre esse tópico. Tente evitar multiposting em diferentes tópicos - já que pode ser considerado spam.",
    heading: "Crie tópicos",
  },
  {
    body:
      "É por isso que excluiremos quaisquer comentários obviamente comerciais ou semelhantes a spam. Isso inclui, mas não se limita a, autopromoção com links para contas sociais, sites, emails, números de telefone; postagem repetitiva do mesmo conteúdo ou links; vender ou distribuir qualquer produto.",
    heading: "Spam não é legal",
  },
  {
    body:
      "Se você tiver sugestões ou perguntas sobre qualquer aspecto da participação aqui, entre em contato com nossa equipe através do formulário de Feedback no aplicativo. Lembre-se de que, ao criar uma conta ou acessar ou usar o aplicativo, você reconhece que aceita e concorda com as Regras da Comunidade.",
    heading: "Feedback",
  },
  {
    body:
      "Acreditamos que a comunidade deva ser um lugar seguro para compartilhar seus pensamentos e experiências. Tirar uma foto da tela com mensagens e posts e compartilhar fora do aplicativo não é uma atitude bem-vinda porque consideramos que todas as usuárias deveriam se sentir seguras para compartilhar suas questões pessoais sem que sejam expostas fora do espaço. Usar o printscreen como ferramenta de exposição de uma usuária pode ser passível de punição (podendo banir a usuária que realizou a ação).",
    heading: "Prints são proibidos porque respeitamos a privacidade",
  },
  {
    body:
      "Para garantir que pessoas de todas as origens se sintam seguras e confortáveis ​​aqui, devemos respeitar as leis locais e internacionais e remover comentários que defendam e promovam atividades ilegais, como drogas, consumo de menores de idade, atividade criminosa, abuso de animais, terrorismo e outros.",
    heading: "Comentários promovendo atividades ilegais serão excluídos",
  },
  {
    body:
      "É por isso que você precisa ter pelo menos 18 anos de idade para usar o aplicativo e contribuir com a nossa comunidade. Sua conta será excluída caso você tenha menos de 18 anos. Você também poderá ser excluída da comunidade caso compartilhe material de menores de 18 anos de idade - não sendo a responsável legal.",
    heading: "Protegemos a segurança das crianças",
  },
  {
    body:
      "Reservamo-nos o direito de excluir comentários inapropriados ou irrelevantes sem notificação prévia. Se o comentário de alguém violar essas regras, nós o excluiremos. Usuárias da Elpis que ignorarem persistentemente as Regras da Comunidade terão acesso restrito ao aplicativo sem aviso prévio.",
    heading: "Protegemos a segurança das crianças",
  },
  {
    body:
      'Você pode denunciar comentários abusivos ou outro comportamento inadequado. Naturalmente, você pode não gostar de tudo que vê. Se você acha que o comentário de alguém é inapropriado ou você percebe comportamento abusivo, use o botão "Denunciar" para enviar um comentário para revisão por nossa equipe de moderação para determinar se há uma violação de nossas regras da comunidade.',
  },
];

const Section = (props: SectionProps) => {
  const { body, heading, subHeading } = props.sectionText;

  return (
    <View style={styles.section}>
      {heading && <Title text={heading} bold small />}
      <Text>
        {subHeading && (
          <Text style={{ fontWeight: "bold" }}>{subHeading} </Text>
        )}
        {body}
      </Text>
    </View>
  );
};

const SectionContainer = ({ sectionList }) => {
  return sectionList.map((item, index) => (
    <Section sectionText={item} key={index} />
  ));
};

const CommunityContent = () => {
  const { title } = labels.communityRules;

  return (
    <View style={styles.container}>
      <Icon name="hands-helping" size={40} colour={colours.navyBlueDark} />
      <Title text={title} large />
      <SectionContainer sectionList={content} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: layout.spacingXL,
  },
  section: {
    marginVertical: layout.spacing,
  },
});

export { CommunityContent };
