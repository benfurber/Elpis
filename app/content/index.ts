export { CommunityContent } from "./community";
export { HowToContent } from "./how-to";
export { OnboardingFeaturesContent } from "./onboarding-features";
export { OnboardingFinalContent } from "./onboarding-final";
