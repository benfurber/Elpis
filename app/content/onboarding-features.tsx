import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";

import { Icon, Title } from "components";
import { labels } from "labels";
import { colours, layout } from "styles";

class OnboardingFeaturesContent extends Component {
  render() {
    const { features } = labels.onboarding;

    const iconColour = colours.navyBlueDark;
    const iconSize = 30;

    return (
      <View>
        <View style={styles.row}>
          <Icon colour={iconColour} name="home" size={iconSize} />
          <Title style={styles.subtitle} text={features.theFeedTitle} />
        </View>
        <Text style={styles.text}>{features.theFeedBody}</Text>

        <View style={styles.row}>
          <Icon colour={iconColour} name="comments" size={iconSize} />
          <Title style={styles.subtitle} text={features.joinDiscussionTitle} />
        </View>
        <Text style={styles.text}>{features.joinDiscussionBody}</Text>

        <Text style={styles.bold}>{features.topicsTitle}</Text>
        <Text style={styles.text}>{features.topicsBody}</Text>

        <Text style={styles.bold}>{features.repliesTitle}</Text>
        <Text style={styles.text}>{features.repliesBody}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  bold: {
    fontWeight: "bold",
  },
  row: {
    alignItems: "center",
    flexDirection: "row",
    marginBottom: layout.spacingL,
    marginTop: layout.spacing,
  },
  subtitle: {
    fontStyle: "italic",
    fontWeight: "normal",
    marginBottom: 0,
    paddingLeft: layout.spacing,
  },
  text: {
    paddingBottom: layout.spacingL,
  },
});

export { OnboardingFeaturesContent };
