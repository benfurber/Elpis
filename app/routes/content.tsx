import { createStackNavigator } from "react-navigation-stack";

import { AddReplyScreen, AddTopicScreen, PostScreen } from "screens";

import { stackConfig } from "./config";

const AddContent = createStackNavigator(
  {
    AddReply: AddReplyScreen,
    AddTopic: AddTopicScreen,
  },
  {
    mode: "modal",
    ...stackConfig,
  },
);

const Content = {
  AddContent,
  Post: {
    path: "post/:id",
    screen: PostScreen,
  },
};

export { Content };
