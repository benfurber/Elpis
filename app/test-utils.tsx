export const mockDateNow = date => {
  const dateToday = new Date(date);
  jest.spyOn(Date, "now").mockImplementation(() => dateToday.getTime());
};
