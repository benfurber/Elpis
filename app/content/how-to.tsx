import React from "react";
import { View } from "react-native";

import { Title } from "components";
import { OnboardingFeaturesContent, OnboardingFinalContent } from "content";
import { labels } from "labels";

const HowToContent = () => {
  const { title } = labels.howTo;

  return (
    <View>
      <Title text={title} large />
      <OnboardingFeaturesContent />
      <OnboardingFinalContent />
    </View>
  );
};

export { HowToContent };
