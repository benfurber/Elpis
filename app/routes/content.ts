import {
  CommunityProfileScreen as CommunityProfile,
  UserProfileScreen as UserProfile,
  WebBrowserScreen as Browser,
} from "screens";

import { AddContent } from "./add-content";
import { Post } from "./post";
import { Reply } from "./reply";

const Content = {
  AddContent,
  Browser,
  CommunityProfile,
  Post,
  Reply,
  UserProfile,
};

export { Content };
