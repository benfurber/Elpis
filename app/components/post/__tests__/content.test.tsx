import "react-native";
import React from "react";
import { shallow } from "enzyme";

import { Content } from "../content";

import { mockDateNow } from "../../../test-utils";

describe("Post<Content>", () => {
  it("renders correctly", () => {
    mockDateNow("2000-01-02");

    const createdAt = new Date("2000-01-01");
    const content = "A string";
    const imagePath = "assets/images/image_post_1.jpg";

    const component = shallow(
      <Content
        createdAt={createdAt}
        content={content}
        imagePath={require(imagePath)}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
