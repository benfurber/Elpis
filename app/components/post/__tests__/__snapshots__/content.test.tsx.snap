// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post<Content> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "borderTopRightRadius": 5,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <DisplayContent
      content="A string"
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(51, 51, 51)",
          "fontStyle": "italic",
        },
        Object {
          "marginVertical": 10,
        },
      ]
    }
  >
    Há um dia
  </Text>
</View>
`;

exports[`Post<Content> with markdown renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.85)",
      "borderTopRightRadius": 5,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <DisplayContent
      content="# h1 Heading 8-

      | Option | Description |

      | ------ | ----------- |

      | data   | path to data files to supply the data that will be passed into templates. |

      | engine | engine to be used for processing templates. Handlebars is the default. |

      | ext    | extension to be used for dest files. |
"
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "rgb(51, 51, 51)",
          "fontStyle": "italic",
        },
        Object {
          "marginVertical": 10,
        },
      ]
    }
  >
    Há um dia
  </Text>
</View>
`;
